Source/controls/plrctrls.cpp

```diff
@@ -941,13 +933,13 @@ void InvMove(AxisDirection dir)
 /**
  * check if hot spell at X Y exists
  */
-bool HSExists(int x, int y)
+bool HSExists(Point target)
 {
 	for (int r = 0; r < speedspellcount; r++) {
-		if (x >= speedspellscoords[r].x - SPLICONLENGTH / 2
-		    && x < speedspellscoords[r].x + SPLICONLENGTH / 2
-		    && y >= speedspellscoords[r].y - SPLICONLENGTH / 2
-		    && y < speedspellscoords[r].y + SPLICONLENGTH / 2) {
+		if (target.x >= speedspellscoords[r].x - SPLICONLENGTH / 2
+		    && target.x < speedspellscoords[r].x + SPLICONLENGTH / 2
+		    && target.y >= speedspellscoords[r].y - SPLICONLENGTH / 2
+		    && target.y < speedspellscoords[r].y + SPLICONLENGTH / 2) {

```

Comment:
Rectangle with a check for `Rectangle.Contains(target)` would be a nice opportunity for cleaning this :)

Reply:
Agreed, though I think it would be more appropriate and easier to review in a separate PR. There are multiple places in this file using Chebyshev distance style checks like this which isn't appropriate given the integer coordinate system for UI elements and the inability to represent the middle of a region with even dimensions.

The checks can be rewritten to `target.WalkingDistance(speedspellscoords[r]) < SPLICONLENGTH` with a bit of refactoring in the callers which is even simpler than Rectangle.Contains, however that obscures the intent so I think the more "wasteful" approach of construction a temporary rectangle would be clearer.